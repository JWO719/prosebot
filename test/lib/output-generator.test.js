const OutputGenerator = require('../../lib/output-generator')
const defaultConfig = require('../../lib/default-config')

describe('OutputGenerator', () => {
  let map, generator

  beforeEach(() => {
    map = new Map()
    const obj = {
      'filename.md': '# This is some text',
      'anotherfile.md': 'this **Has** some Problems. it sure does jason?',
      'cats.md': 'So the cat was stolen.'
    }

    for (const key in obj) {
      map.set(key, obj[key])
    }

    generator = new OutputGenerator(map, defaultConfig)
  })

  describe('static#removeMarkdownFromFiles', () => {
    it('removes markdown from the files and returns the expected map', () => {
      const actual = OutputGenerator.removeMarkdownFromFiles(map)
      expect(actual).toMatchSnapshot()
    })
  })

  describe('#reachConclusion', () => {
    it('returns `failure` if there are any suggestions', () => {
      const results = generator.buildAllResults()
      const actual = generator.reachConclusion(results)
      expect(actual).toBe('failure')
    })

    it('returns `success` if there are no suggestions', () => {
      const actual = generator.reachConclusion(new Map())
      expect(actual).toBe('success')
    })
  })

  describe('#buildSummary', () => {
    it('returns the expected result if the conclusion is `success`', () => {
      const actual = generator.buildSummary('success')
      expect(actual).toBe('No issues have been found, great job!')
    })

    it('returns the expected string if there are suggestions', () => {
      const results = generator.buildAllResults()
      const actual = generator.buildSummary('failure', results)
      expect(actual).toMatchSnapshot()
    })
  })

  describe('#buildAnnotations', () => {
    it('returns the expected array of annotations', () => {
      const results = generator.buildAllResults()
      const actual = generator.buildAnnotations(results)
      expect(actual).toMatchSnapshot()
    })
  })

  describe('#generate', () => {
    it('generates the expected result', () => {
      const actual = generator.generate()
      expect(actual).toMatchSnapshot()
    })
  })
})
