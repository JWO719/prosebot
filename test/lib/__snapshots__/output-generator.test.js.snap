// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutputGenerator #buildAnnotations returns the expected array of annotations 1`] = `
Array [
  Object {
    "annotation_level": "warning",
    "end_line": 1,
    "message": "\\"So\\" adds no meaning",
    "path": "cats.md",
    "start_line": 1,
  },
  Object {
    "annotation_level": "warning",
    "end_line": 1,
    "message": "\\"was stolen\\" may be passive voice",
    "path": "cats.md",
    "start_line": 1,
  },
]
`;

exports[`OutputGenerator #buildSummary returns the expected string if there are suggestions 1`] = `"**2 suggestions** have been found in **1 file**."`;

exports[`OutputGenerator #buildSummary returns the expected string if there is just one 1`] = `"**1 suggestion** has been found in **1 file**."`;

exports[`OutputGenerator #buildSummary returns the expected string with zero results 1`] = `"**0 suggestions** have been found in **0 files**."`;

exports[`OutputGenerator #generate generates the expected result 1`] = `
Object {
  "annotations": Array [
    Object {
      "annotation_level": "warning",
      "end_line": 1,
      "message": "\\"So\\" adds no meaning",
      "path": "cats.md",
      "start_line": 1,
    },
    Object {
      "annotation_level": "warning",
      "end_line": 1,
      "message": "\\"was stolen\\" may be passive voice",
      "path": "cats.md",
      "start_line": 1,
    },
  ],
  "summary": "**2 suggestions** have been found in **1 file**.",
  "title": "See suggestions below.",
}
`;

exports[`OutputGenerator #generate only uses the enabled providers 1`] = `
Object {
  "annotations": Array [],
  "summary": "No issues have been found, great job!",
  "title": "No issues have been found!",
}
`;

exports[`OutputGenerator static#removeMarkdownFromFiles removes markdown from the files and returns the expected map 1`] = `
Map {
  "filename.md" => "This is some text",
  "anotherfile.md" => "this Has some Problems. it sure does Jason?",
  "cats.md" => "So the cat was stolen.",
}
`;
